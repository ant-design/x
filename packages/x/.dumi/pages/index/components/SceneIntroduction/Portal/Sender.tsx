import { AntDesignOutlined, AppstoreOutlined } from '@ant-design/icons';
import { Sender, type SenderProps } from '@ant-design/x';
import tokenData from '@ant-design/x/es/version/token.json';
import { Button, Flex, GetRef } from 'antd';
import { createStyles } from 'antd-style';
import React, { useEffect, useRef, useState } from 'react';
import useLocale from '../../../../../hooks/useLocale';

const locales = {
  cn: {
    function: '功能',
    components: '组件',
    placeholder: '可以问我关于 Ant Design X 的任何问题',
    functionPromptPlaceholder: '请选择一个您想了解功能',
    functionPrompt_1: '请介绍',
    functionPrompt_2: '相关的功能以及写出简单的示例',
    componentsPromptPlaceholder: '请选择一个您想了解的组件',
    componentsPrompt_1: '请介绍Ant Design X 中的',
    componentsPrompt_2: '组件，相关的功能以及写出简单的示例',
  },
  en: {
    function: 'Function',
    components: 'Components',
    placeholder: 'You can ask me anything about Ant Design X',
    functionPromptPlaceholder: 'Please select a function you want to know about',
    functionPrompt_1: 'Please introduce',
    functionPrompt_2: 'the related function and provide a simple example',
    componentsPromptPlaceholder: 'Please select a component you want to know about',
    componentsPrompt_1: 'Please introduce ',
    componentsPrompt_2:
      'component in Ant Design X, its related functions and provide a simple example',
  },
};

export const useStyle = createStyles(({ css, token }) => {
  return {
    sender: css`
      box-sizing:border-box;
      background: linear-gradient(135deg, #ffffff26 14%, #ffffff0d 59%);
      position: relative;
      border: none;
      cursor: pointer;
      border-color:rgba(255,255,255,.1)!important;
      transition: all ${token.motionDurationMid} ${token.motionEaseInOut};
      :hover, :active,:focus-within {
        opacity: 0.85;
        border-color:rgba(255,255,255,.5)!important;
        box-shadow: 0 4px 12px 0 #bfcded33!important;
      }
      .ant-sender-input{
        caret-color:#fff!important;
      }
    `,
  };
});

const CustomizationSender: React.FC<{
  onSubmit: (text: string) => void;
  loading?: boolean;
  abort?: () => void;
}> = (props) => {
  const { styles } = useStyle();
  const [locale] = useLocale(locales);
  const [mergeLoading, setMergeLoading] = useState(false);
  const senderRef = useRef<GetRef<typeof Sender>>(null);
  const [activeKey, setActiveKey] = useState('');
  const options = Object.keys(tokenData) || [];
  options.push('Notification', 'XProvider');
  const SlotInfo: {
    key: string;
    icon: React.ReactNode;
    slotConfig: SenderProps['initialSlotConfig'];
  }[] = [
    {
      icon: <AppstoreOutlined />,
      key: 'function',
      slotConfig: [
        { type: 'text', value: locale.functionPrompt_1 },
        {
          type: 'select',
          key: 'search_function',
          props: {
            defaultValue: 'X-SDK',
            options: ['X-SDK', 'X-Markdown', 'X组件'],
            placeholder: locale.functionPromptPlaceholder,
          },
        },
        { type: 'text', value: locale.functionPrompt_2 },
      ],
    },
    {
      key: 'components',
      icon: <AntDesignOutlined />,
      slotConfig: [
        { type: 'text', value: locale.componentsPrompt_1 },
        {
          type: 'select',
          key: 'components',
          props: {
            options,
            placeholder: locale.componentsPromptPlaceholder,
          },
        },
        { type: 'text', value: locale.componentsPrompt_2 },
      ],
    },
  ];

  useEffect(() => {
    if (props?.loading !== undefined) {
      setMergeLoading(props?.loading);
    }
  }, [props.loading]);
  useEffect(() => {
    senderRef.current?.focus();
  }, []);

  const onClick = (key: string) => {
    setActiveKey(key);
  };

  return (
    <Sender
      ref={senderRef}
      placeholder={locale.placeholder}
      loading={mergeLoading}
      className={styles.sender}
      onCancel={props.abort}
      key={activeKey}
      styles={{
        input: {
          fontSize: 15,
        },
      }}
      autoSize={{ minRows: 2, maxRows: 2 }}
      onSubmit={(value) => {
        if (!mergeLoading) {
          setMergeLoading(true);
          props.onSubmit(value);
          setActiveKey('');
          senderRef.current?.clear?.();
        }
      }}
      initialSlotConfig={SlotInfo.find(({ key }) => key === activeKey)?.slotConfig}
      suffix={false}
      footer={(_, info) => {
        const { LoadingButton } = info.components;

        return (
          <Flex justify="space-between" align="center">
            <Flex gap="small">
              {SlotInfo.map(({ key, icon }) => (
                <Sender.Switch
                  key={key}
                  value={activeKey === key}
                  checkedChildren={locale?.[key as keyof typeof locale]}
                  unCheckedChildren={locale?.[key as keyof typeof locale]}
                  onChange={(checked: boolean) => {
                    if (checked) {
                      onClick(key);
                    } else {
                      onClick('');
                    }
                  }}
                  icon={icon}
                />
              ))}
            </Flex>
            {mergeLoading ? (
              <LoadingButton />
            ) : (
              <Button
                type="text"
                style={{ padding: 0 }}
                onClick={() => {}}
                icon={
                  <img
                    alt="send"
                    loading="lazy"
                    src="https://mdn.alipayobjects.com/huamei_iwk9zp/afts/img/A*4e5sTY9lU3sAAAAAAAAAAAAADgCCAQ/original"
                  />
                }
              />
            )}
          </Flex>
        );
      }}
    />
  );
};

export default CustomizationSender;
