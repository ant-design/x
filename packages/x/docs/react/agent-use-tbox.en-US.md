---
group:
  order: 2
  title: Agent Integration
title: Tbox
order: 1
---

This guide introduces how to integrate Tbox agent services into applications built with Ant Design X.

## Related Documentation

- Tbox Open Platform Official Website - [https://tbox.cn/open](https://tbox.cn/open)
- Tbox Open Platform Overview - [https://alipaytbox.yuque.com/sxs0ba/doc/tbox_open_overview](https://alipaytbox.yuque.com/sxs0ba/doc/tbox_open_overview)
- Authorization Management - [https://alipaytbox.yuque.com/sxs0ba/doc/tbox_open_token](https://alipaytbox.yuque.com/sxs0ba/doc/tbox_open_token)
- OpenAPI - [https://alipaytbox.yuque.com/sxs0ba/doc/tbox_openapi_overview](https://alipaytbox.yuque.com/sxs0ba/doc/tbox_openapi_overview)

### tbox-nodejs-sdk

```ts
import { TboxClient } from 'tbox-nodejs-sdk';

const client = new TboxClient({
  httpClientConfig: {
    authorization: 'Tbox-your-token-xxx',
  },
});

const stream = client.chat({
  appId: 'your-app-id',
  query: '今天杭州天气怎么样？',
  userId: 'user123',
});

stream.on('data', (data) => {
  console.log('Received data:', data);
});

stream.on('end', () => {
  console.log('Stream ended');
});

stream.on('error', (error) => {
  console.error('Stream error:', error);
});
```

## Integrate tbox-nodejs-sdk with X SDK

Using URL to integrate agents is a basic capability provided by X SDK. For details, see [X SDK](/sdks/introduce). For the complete Tbox template, see [Template - Tbox](/docs/playground/agent-tbox).

### Example

<code src="./demo/tbox.tsx" title="Integrate with X SDK"></code>
