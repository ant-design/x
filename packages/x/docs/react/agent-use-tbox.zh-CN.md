---
group:
  order: 2
  title: 智能体接入
title: 百宝箱
order: 1
---

这篇指南将介绍如何在使用 Ant Design X 搭建的应用中接入百宝箱智能体服务。

## 相关文档

- 百宝箱开放平台官网 - [https://tbox.cn/open](https://tbox.cn/open)
- 百宝箱开放平台概述 - [https://alipaytbox.yuque.com/sxs0ba/doc/tbox_open_overview](https://alipaytbox.yuque.com/sxs0ba/doc/tbox_open_overview)
- 授权管理 - [https://alipaytbox.yuque.com/sxs0ba/doc/tbox_open_token](https://alipaytbox.yuque.com/sxs0ba/doc/tbox_open_token)
- OpenAPI - [https://alipaytbox.yuque.com/sxs0ba/doc/tbox_openapi_overview](https://alipaytbox.yuque.com/sxs0ba/doc/tbox_openapi_overview)

### tbox-nodejs-sdk

```ts
import { TboxClient } from 'tbox-nodejs-sdk';

const client = new TboxClient({
  httpClientConfig: {
    authorization: 'TBox-your-token-xxx',
  },
});

const stream = client.chat({
  appId: 'your-app-id',
  query: '今天杭州天气怎么样？',
  userId: 'user123',
});

stream.on('data', (data) => {
  console.log('Received data:', data);
});

stream.on('end', () => {
  console.log('Stream ended');
});

stream.on('error', (error) => {
  console.error('Stream error:', error);
});
```

## 使用 X SDK 接入 tbox-nodejs-sdk

使用URL接入智能体是 X SDK提供的基础能力，详情请查看[X SDK](/sdks/introduce-cn)，百宝箱完整样板间请查看[样板间-百宝箱](/docs/playground/agent-tbox-cn)。

### 示例

<code src="./demo/tbox.tsx" title="使用X SDK接入"></code>
