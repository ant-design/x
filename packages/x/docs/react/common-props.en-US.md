---
group:
  order: 4
  title: Advanced Usage
title: Common Props
order: 1
tag: Updated
---

> Tips: The following common props apply to most Ant Design X components. Components that do not support them will be specified separately.

| Property      | Description                         | Type                         | Default |
| ------------- | ----------------------------------- | ---------------------------- | ------- |
| prefixCls     | Unified style prefix                | string                       | -       |
| style         | Custom style                        | CSSProperties                | -       |
| className     | Custom class name                   | string                       | -       |
| styles        | Semantic style                      | Record<string,CSSProperties> | -       |
| classNames    | Semantic custom class name          | Record<string,string>        | -       |
| rootClassName | Class name on the outermost element | string                       | -       |
