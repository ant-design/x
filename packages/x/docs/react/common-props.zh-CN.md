---
group:
  order: 4
  title: 进阶使用
title: 通用属性
order: 1
---

> Tips: 以下通用属性适用于 Ant Design X 大部分组件，不支持的组件会单独说明。

| 参数          | 说明                         | 类型                         | 默认值 |
| ------------- | ---------------------------- | ---------------------------- | ------ |
| prefixCls     | 统一样式前缀                 | string                       | -      |
| style         | 自定义样式                   | CSSProperties                | -      |
| className     | 自定义类名                   | string                       | -      |
| styles        | 语义化样式                   | Record<string,CSSProperties> | -      |
| classNames    | 语义化自定义类名             | Record<string,string>        | -      |
| rootClassName | 添加在组件最外层的 className | string                       | -      |
