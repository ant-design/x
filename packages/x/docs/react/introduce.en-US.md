---
order: 0
title: Ant Design X
subtitle: ï½œSolution for Intelligent Interface Development
---

Ant Design X is a comprehensive toolkit for AI applications, integrating a UI component library, streaming Markdown rendering engine, and AI SDK, providing developers with a complete toolchain for building next-generation AI-driven applications.

**`@ant-design/x` - Intelligent Interface Framework**

A React UI library based on the Ant Design system, designed for AI-driven interfaces. Out-of-the-box smart conversation components, seamless API integration, and rapid smart interface development. For details, click [here](/components/introduce-en/).

**`@ant-design/x-markdown` - High-performance Streaming Rendering Engine**

An optimized Markdown rendering solution for streaming content, with powerful extension capabilities. Supports formulas, code highlighting, mermaid charts, and delivers excellent performance for smooth content display. For details, click [here](/markdowns/introduce-en).

**`@ant-design/x-sdk` - AI Conversation Data Flow Management**

Provides a complete set of tool APIs, out-of-the-box AI conversation data flow management, simplifying development and improving efficiency. For details, click [here](/markdowns/introduce-en).

<div class="pic-plus">
  <img width="150" src="https://mdn.alipayobjects.com/huamei_iwk9zp/afts/img/A*eco6RrQhxbMAAAAAAAAAAAAADgCCAQ/original"/>
  <span>+</span>
  <img width="160" src="https://gw.alipayobjects.com/zos/antfincdn/aPkFc8Sj7n/method-draw-image.svg"/>
  <span>+</span>
  <img width="200" src="https://mdn.alipayobjects.com/huamei_lkxviz/afts/img/2v_BT6g_DFUAAAAAVEAAAAgADtFMAQFr/original"/>
</div>

---

![](https://mdn.alipayobjects.com/huamei_iwk9zp/afts/img/A*UAEeSbJfuM8AAAAAAAAAAAAADgCCAQ/fmt.webp)

## Who is using

Ant Design X is widely used in AI-driven user interfaces within Ant Group. If your company and products use Ant Design X, feel free to leave a comment [here](https://github.com/ant-design/x/issues/126).

## How to contribute

Before participating in any form, please read the [CONTRIBUTING.md](https://github.com/ant-design/ant-design/blob/master/.github/CONTRIBUTING.md) first. If you want to contribute, feel free to submit a [Pull Request](https://github.com/ant-design/ant-design/pulls) or [report a bug](http://new-issue.ant.design/).

> Highly recommend reading [How To Ask Questions The Smart Way](https://github.com/ryanhanwu/How-To-Ask-Questions-The-Smart-Way), [How to Ask a Question in Open Source Community](https://github.com/seajs/seajs/issues/545), [How to Report Bugs Effectively](http://www.chiark.greenend.org.uk/~sgtatham/bugs.html), and [How to Submit Unsolvable Issues to Open Source Projects](https://zhuanlan.zhihu.com/p/25795393). Well-written questions are more likely to get help.

## Community Support

If you encounter any issues while using Ant Design X, you can seek help through the following channels. We also encourage experienced users to assist newcomers via these platforms.

When asking questions on GitHub Discussions, it's recommended to use the `Q&A` tag.

1. [GitHub Discussions](https://github.com/ant-design/x/discussions)
2. [GitHub Issues](https://github.com/ant-design/x/issues)
