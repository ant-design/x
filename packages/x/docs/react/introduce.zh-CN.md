---
order: 0
title: Ant Design X
subtitle: ｜智能界面开发的解决方案
---

Ant Design X 是一款AI应用复合工具集，融合了 UI 组件库、流式 Markdown 渲染引擎和 AI SDK，为开发者提供构建下一代 AI 驱动应用的完整工具链。

**`@ant-design/x` - 智能界面构建框架**

基于 Ant Design 设计体系的 React UI 库、专为 AI 驱动界面设计，开箱即用的智能对话组件、无缝集成 API 服务，快速搭建智能应用界面，查看详情请点击[这里](/components/introduce-cn/)。

**`@ant-design/x-markdown` - 高性能流式渲染引擎**

专为流式内容优化的 Markdown 渲染解决方案、强大的扩展能力，支持公式、代码高亮、mermaid 图表等极致性能表现，确保流畅的内容展示体验。查看详情请点击[这里](/markdowns/introduce-cn)。

**`@ant-design/x-sdk` - AI 对话数据流管理**

提供完整的工具 API 集合、开箱即用的 AI 对话应用数据流管理、简化开发流程，提升开发效率。查看详情请点击[这里](/markdowns/introduce-cn)。

<div class="pic-plus">
  <img width="150" src="https://mdn.alipayobjects.com/huamei_iwk9zp/afts/img/A*eco6RrQhxbMAAAAAAAAAAAAADgCCAQ/original"/>
  <span>+</span>
  <img width="160" src="https://gw.alipayobjects.com/zos/antfincdn/aPkFc8Sj7n/method-draw-image.svg"/>
    <span>+</span>
  <img width="200" src="https://mdn.alipayobjects.com/huamei_lkxviz/afts/img/2v_BT6g_DFUAAAAAVEAAAAgADtFMAQFr/original"/>

</div>

---

![](https://mdn.alipayobjects.com/huamei_iwk9zp/afts/img/A*UAEeSbJfuM8AAAAAAAAAAAAADgCCAQ/fmt.webp)

## 谁在使用

Ant Design X 广泛用于蚂蚁集团内由 AI 驱动的用户交互界面。如果你的公司和产品使用了 Ant Design X，欢迎到 [这里](https://github.com/ant-design/x/issues/126) 留言。

## 如何贡献

在任何形式的参与前，请先阅读 [贡献者文档](https://github.com/ant-design/ant-design/blob/master/.github/CONTRIBUTING.md)。如果你希望参与贡献，欢迎提交 [Pull Request](https://github.com/ant-design/ant-design/pulls)，或给我们 [报告 Bug](http://new-issue.ant.design/)。

> 强烈推荐阅读 [《提问的智慧》](https://github.com/ryanhanwu/How-To-Ask-Questions-The-Smart-Way)、[《如何向开源社区提问题》](https://github.com/seajs/seajs/issues/545) 和 [《如何有效地报告 Bug》](http://www.chiark.greenend.org.uk/%7Esgtatham/bugs-cn.html)、[《如何向开源项目提交无法解答的问题》](https://zhuanlan.zhihu.com/p/25795393)，更好的问题更容易获得帮助。

## 社区互助

如果您在使用的过程中碰到问题，可以通过下面几个途径寻求帮助，同时我们也鼓励资深用户通过下面的途径给新人提供帮助。

通过 GitHub Discussions 提问时，建议使用 `Q&A` 标签。

1. [GitHub Discussions](https://github.com/ant-design/x/discussions)
2. [GitHub Issues](https://github.com/ant-design/x/issues)
