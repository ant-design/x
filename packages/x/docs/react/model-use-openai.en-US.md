---
group:
  title: Model Integration
  order: 1
title: OpenAI
order: 0
---

This guide introduces how to integrate OpenAI model services into applications built with Ant Design X. For details, see [X SDK](/sdks/introduce)

## Using OpenAI API

Equivalent to integrating model inference services compatible with OpenAI. Refer to [Model Integration - Qwen](/docs/react/model-use-qwen)

## Using openai-node

Usually, openai-node is used in Node environments. If you use it in the browser, you need to enable `dangerouslyAllowBrowser`.

> Note: `dangerouslyAllowBrowser` has security risks. See the official documentation for details: [openai-node requirements](https://github.com/openai/openai-node?tab=readme-ov-file#requirements).

```tsx
import { Bubble, BubbleListProps, Sender } from '@ant-design/x';
import {
  AbstractXRequestClass,
  OpenAIChatProvider,
  SSEFields,
  useXChat,
  XModelMessage,
  XModelParams,
  XRequestOptions,
} from '@ant-design/x-sdk';
import { Flex } from 'antd';
import OpenAI from 'openai';
import React, { useState } from 'react';

type OutputType = Partial<Record<SSEFields, any>>;
type InputType = XModelParams;

class OpenAiRequest<
  Input extends InputType = InputType,
  Output extends OutputType = OutputType,
> extends AbstractXRequestClass<Input, Output> {
  client: any;
  stream: OpenAI | undefined;

  _isTimeout = false;
  _isStreamTimeout = false;
  _isRequesting = false;

  constructor(baseURL: string, options: XRequestOptions<Input, Output>) {
    super(baseURL, options);
    this.client = new OpenAI({
      apiKey: 'OPENAI_API_KEY',
      dangerouslyAllowBrowser: true,
    });
  }
  get asyncHandler(): Promise<any> {
    return Promise.resolve();
  }
  get isTimeout(): boolean {
    return this._isTimeout;
  }
  get isStreamTimeout(): boolean {
    return this._isStreamTimeout;
  }
  get isRequesting(): boolean {
    return this._isRequesting;
  }
  get manual(): boolean {
    return true;
  }
  async run(input?: Input | undefined): Promise<void> {
    const { callbacks } = this.options;
    try {
      await this.client.responses.create({
        model: 'gpt-4o',
        input: input?.messages?.[0]?.content || '',
        stream: true,
      });

      // Please implement stream data update logic based on response
    } catch (error: any) {
      callbacks?.onError(error);
    }
  }
  abort(): void {
    // Please implement abort based on OpenAI
  }
}

const provider = new OpenAIChatProvider<XModelMessage, InputType, OutputType>({
  request: new OpenAiRequest('OPENAI', {}),
});

const Demo: React.FC = () => {
  const [content, setContent] = useState('');
  const { onRequest, messages, requesting, abort } = useXChat({
    provider,
    requestPlaceholder: () => {
      return {
        content: 'loading...',
        role: 'assistant',
      };
    },
    requestFallback: (_, { error }) => {
      if (error.name === 'AbortError') {
        return {
          content: 'Request is aborted',
          role: 'assistant',
        };
      }
      return {
        content: error?.toString(),
        role: 'assistant',
      };
    },
  });

  const items = messages.map(({ message, id }) => ({
    key: id,
    ...message,
  }));

  const role: BubbleListProps['role'] = {
    assistant: {
      placement: 'start',
    },
    user: { placement: 'end' },
  };

  return (
    <Flex
      vertical
      justify="space-between"
      style={{
        height: 400,
        padding: 16,
      }}
    >
      <Bubble.List role={role} items={items} />
      <Sender
        value={content}
        onChange={setContent}
        loading={requesting}
        onCancel={abort}
        onSubmit={(val) => {
          onRequest({
            messages: [{ role: 'user', content: val }],
          });
          setContent('');
        }}
      />
    </Flex>
  );
};

export default Demo;
```

## Example

<code src="./demo/openai-node.tsx" title="Integrate openai" description="This example only shows the logic reference for integrating openai with X SDK. Model data is not processed, please fill in the correct apiKey for data debugging." compact iframe="450"></code>
