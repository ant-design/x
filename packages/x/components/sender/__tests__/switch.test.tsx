import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';
import Sender from '../index';

describe('Sender.Switch', () => {
  mountTest(() => <Sender.Switch />);
  rtlTest(() => <Sender.Switch />);

  it('should support controlled mode', () => {
    const Demo = () => {
      const [checked, setChecked] = React.useState(false);
      return (
        <Sender.Switch
          value={checked}
          onChange={setChecked}
          checkedChildren="ON"
          unCheckedChildren="OFF"
        />
      );
    };
    const { getByText } = render(<Demo />);
    // Initial state is OFF
    expect(getByText('OFF')).toBeInTheDocument();
    // 点击切换
    fireEvent.click(getByText('OFF'));
    expect(getByText('ON')).toBeInTheDocument();
  });

  it('should support uncontrolled mode', () => {
    const { getByText } = render(<Sender.Switch checkedChildren="ON" unCheckedChildren="OFF" />);
    expect(getByText('OFF')).toBeInTheDocument();
    fireEvent.click(getByText('OFF'));
    expect(getByText('ON')).toBeInTheDocument();
  });

  it('should support disabled', () => {
    const onChange = jest.fn();
    const { container } = render(<Sender.Switch disabled onChange={onChange} />);
    // Should not trigger onChange when disabled
    fireEvent.click(container?.querySelector('.ant-sender-switch') as Element);
    expect(onChange).not.toHaveBeenCalled();
  });

  it('should support loading', () => {
    const { container } = render(<Sender.Switch loading />);
    expect(container.querySelector('.anticon-loading')).toBeTruthy();
  });

  it('should support icon', () => {
    const { container } = render(<Sender.Switch icon={<span data-testid="icon">icon</span>} />);
    expect(container.querySelector('[data-testid="icon"]')).toBeTruthy();
  });

  it('should render checkedChildren and unCheckedChildren', () => {
    const { getByText } = render(<Sender.Switch checkedChildren="YES" unCheckedChildren="NO" />);
    expect(getByText('NO')).toBeInTheDocument();
    fireEvent.click(getByText('NO'));
    expect(getByText('YES')).toBeInTheDocument();
  });
});
