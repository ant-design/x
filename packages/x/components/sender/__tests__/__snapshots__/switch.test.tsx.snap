// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Sender.Switch rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-sender ant-sender-switch css-var-root ant-sender-rtl"
>
  <button
    class="ant-btn css-var-root ant-btn-default ant-btn-color-default ant-btn-variant-outlined ant-btn-rtl ant-sender-switch-content"
    type="button"
  />
</div>
`;
